src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,36 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User is null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("User's login is null");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("User's password is null");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("User's age is null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with this login is already in the storage");
+        }
+        if (user.getLogin().length() >= 6
+                && user.getPassword().length() >= 6
+                && user.getAge() >= 18) {
+            storageDao.add(user);
+            return user;
+        } else {
+            throw new RegistrationException("Wrong user data");
+        }
```

Comment:
no else after return needed
Better invert condition, throw in if
exception message is  uninformative